pisces/data_sets.py

```diff
@@ -291,16 +293,19 @@ def get_filename(self, feature: str, id: str) -> Path | None:
     def get_feature_path(self, feature: str) -> Path:
         return self.path.joinpath(self.FEATURE_PREFIX + feature)
     
-    def _extract_ids(self, files: List[str]) -> List[str]:
-        return IdExtractor().extract_ids(files)
+    def _extract_ids(self, files: List[str],
+                     id_template: str | None=None,
+                     id_symbol: str="<<ID>>") -> List[str]:
```

Comment:
Since this is a private method, I think it's better to have no defaults. Only this class should be using it, let's use it fully.

Also, having 2 functions that call each other, with identical param names, both of which have a default (in this case, the same), seems to be more annoying than it's worth in the long run. I'd suggest adding only a default param at the most common call site, so I guess that's `DataSetObject.find_data_sets`.

One more thing that I'm guessing is that `id_template` and `id_symbol` probably can change per data set... let's discuss more on our call.